browser/app/js/browser/OpenIDLogin.js

```diff
@@ -0,0 +1,173 @@
+/*
+ * MinIO Cloud Storage (C) 2019 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from "react"
+import { connect } from "react-redux"
+import logo from "../../img/logo.svg"
+import Alert from "../alert/Alert"
+import * as actionsAlert from "../alert/actions"
+import InputGroup from "./InputGroup"
+import web from "../web"
+import { Redirect } from "react-router-dom"
+import qs from "query-string"
+import SuperAgent from "superagent-es6-promise"
+import { getRandomString } from "../utils"
+import storage from "local-storage-fallback"
+import jwtDecode from "jwt-decode"
+
+export class OpenIDLogin extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      clientID: "",
+      OpenIDConfigURL: undefined
+    }
+    this.clientIDChange = this.clientIDChange.bind(this)
+    this.handleSubmit = this.handleSubmit.bind(this)
+  }
+
+  clientIDChange(e) {
+    this.setState({
+      clientID: e.target.value
+    })
+  }
+
+  handleSubmit(event) {
+    event.preventDefault()
+    const { showAlert } = this.props
+    let message = ""
+    if (this.state.clientID === "") {
+      message = "Client ID cannot be empty"
+    }
+    if (message) {
+      showAlert("danger", message)
+      return
+    }
+    if (this.state.OpenIDConfigURL) {
+      let req = SuperAgent.get(this.state.OpenIDConfigURL)
+      req.send().then(res => {
+        const redirectURI = window.location.href.split("#")[0]
+        var params = new URLSearchParams()
+        params.set("response_type", "id_token")
+        params.set("scope", "openid")
+        params.set("client_id", this.state.clientID)
+        params.set("redirect_uri", redirectURI)
+
+        // Store nonce in localstorage to check again after the redirect
+        const nonce = getRandomString(16)
+        params.set("nonce", nonce)
+        storage.setItem("openIDKey", nonce)
+
+        const authURL = `${
+          res.body.authorization_endpoint
+        }?${params.toString()}`
+        window.location = authURL
+      })
+    }
+  }
+
+  componentWillMount() {
```

Comment:
@kanagarajkm, should we move this code block inside `componentDidMount` as this lifecycle method will eventually be deprecated if we decide to bump up the React version? The same thing could also be said for the usage of `componentWillMount` on Login.js as well.